### PR TITLE
Sim2h space data remote debugging

### PR DESCRIPTION
**Pull request by lucksus**
## PR summary
Adds a `WireMessge::Debug` and `WireMessage::DebugResponse(BTreeMap<SpaceHash, String>)` to retrieve all space data as JSON serialized string.

`hc` can request this debug state dump and write it to file(s) (each space data into a separate one):
```
[nix-shell:~/hc]$ hc sim2h-client --message debug --url ws://localhost:9000
url: ws://localhost:9000
message: debug
connecting to: ws://127.0.0.1:9000/
Generated agent id: HcSCiDdSTjGaajts3zqY9WqC4zfJhnuhs8m4R6rC35RwObu8Uc6seVaYWe9eija
Await successfull
Sending wire message to sim2h: Debug
Ack(3079098903191232539)
Got DebugResponse for 1 spaces.
Writing Sim2h state dump for space QmXghcjdFQ7vEvihuBVQdiQBATGnRo6qX5wNxrq6HdNH9i to file: QmXghcjdFQ7vEvihuBVQdiQBATGnRo6qX5wNxrq6HdNH9i.json
```

## testing/benchmarking notes

( if any manual testing or benchmarking was/should be done, add notes and/or screenshots here )

## followups

**Make this optional (per command line parameter / configuration) such that public sim2h servers don't offer potential sensitive information to everybody**

## changelog

- [x] if this is a code change that effects some consumer (e.g. zome developers) of holochain core,  then it has been added to [our between-release changelog](https://github.com/holochain/holochain-rust/blob/develop/CHANGELOG-UNRELEASED.md) with the format 

```markdown
- summary of change [PR#1234](https://github.com/holochain/holochain-rust/pull/1234)
```

## documentation

- [ ] this code has been documented according to our [docs checklist](https://hackmd.io/@freesig/Hk9AmKJNS)

### CODE REVIEW

**Comment by zippy** [comment]
I think this is going to need dumping just the response to a file maybe JSON encoded to be usefull?

**Comment by lucksus** [comment]
> I think this is going to need dumping just the response to a file maybe JSON encoded to be usefull?

Yep, mainly why I left this as WIP. The current plain debug output is hard to read..

File: crates/cli/src/cli/sim2h_client.rs [hunk 1]
```diff
@@ -41,6 +45,7 @@ pub fn sim2h_client(url_string: String, message_string: String) -> Result<(), St
         "ping" => WireMessage::Ping,
         "hello" => WireMessage::Hello(WIRE_VERSION),
         "status" => WireMessage::Status,
+        "debug" => WireMessage::Debug,
         _ => {
             return Err(format!(
                 "expecting 'ping' or 'status' for message, got: {}",
```
Comment (zippy) [inline crates/cli/src/cli/sim2h_client.rs]:
```suggestion
                "expecting 'ping', 'status' or 'debug' for message, got: {}",
```

**Comment by zippy** [comment]
Fails if message debug data is bigger than max tungstenite message size!

**Pull request merged**